datajunction-clients/python/datajunction/builder.py

```diff
@@ -112,6 +112,11 @@ def create_node(
 
         try:
             existing_node_dict = self._get_node(name)
+            if type_ == models.NodeType.CUBE:
+                cube_dict = self._get_cube(name)
+                if "name" in cube_dict:
```

Comment:
This check seems unnecessary. Or am I missing something?

Reply:
`_get_node` and `_get_cube` hit two different API endpoints, `/nodes/` and `/cubes/` respectively. Only the latter returns the necessary fields, specifically `metrics` and `dimensions`

Reply:
I understand that, but I don't understand that if `self._get_cube(name)` returns w/o failure and returns some dict, do we really need to check if that dict has a key called "name" ?

Reply:
Yeah, I think I only added this because the unit tests don't raise an error. I've removed it since it works for the actual client.

Reply:
The things we do for unit tests :)